Src/VsVimShared/Implementation/Settings/VimApplicationSettings.cs

```diff
@@ -38,6 +38,7 @@ internal sealed class VimApplicationSettings : IVimApplicationSettings
 
         private readonly WritableSettingsStore _settingsStore;
         private readonly IProtectedOperations _protectedOperations;
+        private readonly ISettingsCache _settingsCache;

```

Comment:
Rather than having consumers deal with both `ISettingsCache` and `WritableSettingsStore` did you just consider adding an abstraction over `WritableSettingsStore`? Essentially say `CachableSettingsStore` that has the same interface as `WritableSettingsStore` but does the in memory caching you wrote?